Assignment 2/src/main/java/com/nuclei/assignment/service/inputvalidation/InputValidationImpl.java

```diff
@@ -0,0 +1,159 @@
+package com.nuclei.assignment.service.inputvalidation;
+
+import com.nuclei.assignment.constants.ExceptionsConstantsUtils;
+import com.nuclei.assignment.constants.StringConstantsUtils;
+import com.nuclei.assignment.enums.Courses;
+import com.nuclei.assignment.enums.SortingOrder;
+import com.nuclei.assignment.exception.CustomException;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * User Validation Implementation containing method to validate different properties.
+ */
+public class InputValidationImpl implements InputValidation {
+  
+  /**
+   * The Logger.
+   */
+  private final Log logger = LogFactory.getLog(InputValidationImpl.class);
+  
+  @Override
+  public String validateName(final String name) throws CustomException {
+    checkDataIsNull(name, "Name");
+    if (name.isEmpty()) {
+      logger.error(ExceptionsConstantsUtils.INVALID_NAME);
+      throw new CustomException(ExceptionsConstantsUtils.INVALID_NAME);
+    }
+    return name;
+  }
+  
+  @Override
+  public String validateAddress(final String address) throws CustomException {
+    checkDataIsNull(address, "Address");
+    if (address.isEmpty()) {
+      logger.error(ExceptionsConstantsUtils.INVALID_ADDRESS);
+      throw new CustomException(ExceptionsConstantsUtils.INVALID_ADDRESS);
+    }
+    return address;
+  }
+  
+  @Override
+  public int validateAge(final String age) throws CustomException {
```

Comment:
These functions are doing more than validations, either think of separating validation from parsing part.

Reply:
Done